### PR TITLE
Update BOUT++ to latest

### PR DESCRIPTION
**Pull request by bendudson**
Fixes compilation issues and upgrades BOUT++ to the `next` branch. This branch includes some important updates, including the `beuler` solver which can be significantly faster at finding steady state solutions than other time integrators.

Also adds a convergence test, automated using Github actions

### CODE REVIEW

File: tests/case-01/runtest [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+# Python script to run and analyse a self-convergence test
+
+from __future__ import division
+from __future__ import print_function
+
+try:
+  from builtins import str
+except:
+  pass
```
Comment (dschwoerer) [inline tests/case-01/runtest]: All of this can go if it is for python3

File: tests/case-01/runtest [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+    print("Running with " + args)
+
+    # Delete old data
+    shell("rm data/BOUT.dmp.*.nc")
+
+    # Command to run
+    cmd = "../../sd1d "+args
+    # Launch using MPI
+    s, out = launch(cmd, nproc=nproc, pipe=True)
```
Comment (dschwoerer) [inline tests/case-01/runtest]: use `launch_safe` instead?

File: tests/case-01/runtest [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+except:
+  pass
+
+from boututils.run_wrapper import shell, launch, getmpirun
+from boutdata.collect import collect
+
+from numpy import sqrt, max, abs, mean, array, log, concatenate
+
+# List of NY values to use
+nylist = [20, 40, 80, 160, 320]
```
Comment (dschwoerer) [inline tests/case-01/runtest]: only the last 3 ones are actually used in the test, not sure what the time overhead is though ...

**Review by dschwoerer — COMMENTED** [review]
`black` for the runtest would be nice :-)

**Comment by bendudson** [comment]
Thanks @dschwoerer for all the suggestions! I think I've implemented them all now

**Pull request merged**